File: phoenix-platform/pkg/models/experiment.go
package models

import (
	"time"
)

// Experiment represents a Phoenix experiment
type Experiment struct {
	ID                string            `json:"id" db:"id"`
	Name              string            `json:"name" db:"name"`
	Description       string            `json:"description" db:"description"`
	BaselinePipeline  string            `json:"baseline_pipeline" db:"baseline_pipeline"`
	CandidatePipeline string            `json:"candidate_pipeline" db:"candidate_pipeline"`
	Status            string            `json:"status" db:"status"`
	TargetNodes       map[string]string `json:"target_nodes" db:"target_nodes"`
	CreatedAt         time.Time         `json:"created_at" db:"created_at"`
	UpdatedAt         time.Time         `json:"updated_at" db:"updated_at"`
	StartedAt         *time.Time        `json:"started_at" db:"started_at"`
	CompletedAt       *time.Time        `json:"completed_at" db:"completed_at"`
}

// ExperimentStatus represents possible experiment states
const (
	ExperimentStatusPending   = "pending"
	ExperimentStatusRunning   = "running"
	ExperimentStatusCompleted = "completed"
	ExperimentStatusFailed    = "failed"
	ExperimentStatusStopped   = "stopped"
)

File: phoenix-full-backup-20250526-183503/dashboard/src/types/experiment.ts
export interface Experiment {
  id: string
  name: string
  description?: string
  spec?: ExperimentSpec
  status: ExperimentStatus
  owner?: string
  createdAt: string
  updatedAt: string
  startedAt?: string
  completedAt?: string
  results?: ExperimentResults
}

export type ExperimentStatus = 
  | 'pending' 
  | 'initializing' 
  | 'running' 
  | 'analyzing' 
  | 'completed' 
  | 'failed' 
  | 'cancelled'

export interface ExperimentSpec {
  baseline: PipelineConfig
  candidate: PipelineConfig
  targetHosts?: string[]
  duration?: string
  loadProfile?: 'realistic' | 'high-cardinality' | 'high-churn'
  successCriteria?: SuccessCriteria
}

export interface SuccessCriteria {
  minCardinalityReduction: number
  maxCostIncrease: number
  maxLatencyIncrease: number
  minCriticalProcessRetention: number
}

export interface PipelineConfig {
  name?: string
  receivers?: string[]
  processors?: ProcessorConfig[]
  exporters?: string[]
}

export interface ProcessorConfig {
  type: string
  config?: Record<string, any>
}

export interface ExperimentResults {
  baselineMetrics: MetricsSummary
  candidateMetrics: MetricsSummary
  costReduction: number
  cardinalityReduction: number
  summary: string
  recommendation?: 'baseline' | 'candidate'
}

export interface MetricsSummary {
  cardinality: number
  cpuUsage: number
  memoryUsage: number
  networkTraffic: number
  dataPointsPerSecond?: number
  uniqueProcesses?: number
}

export interface MetricsData {
  timestamp: number[]
  values: number[]
  metric: string
  variant: 'baseline' | 'candidate'
}

export interface AnalysisResult {
  experimentId: string
  status: 'pending' | 'in-progress' | 'completed' | 'failed'
  startTime: string
  endTime?: string
  comparison: {
    cardinalityReduction: number
    costSavings: number
    criticalProcessRetention: number
    performanceImpact: {
      cpuOverhead: number
      memoryOverhead: number
      latencyIncrease: number
    }
  }
  recommendation: {
    variant: 'baseline' | 'candidate'
    confidence: number
    reasons: string[]
  }
}

export interface CreateExperimentData {
  name: string
  description?: string
  spec: ExperimentSpec
}